ap_git/core.py

```diff
@@ -648,3 +650,93 @@ def shallow_clone_at_commit_from_local(source: str,
         source_repo.__branch_delete(branch_name=temp_branch_name, force=True)
 
         return cloned_repo
+
+    @staticmethod
+    def from_path(path: str,
+                  clone_from: str = None,
+                  recurse_submodules: bool = False,
+                  shallow_submodules: bool = False,
+                  clone_if_broken: bool = False,
+                  clone_if_not_found: bool = False) -> "GitRepo":
+        """
+        Initialise a GitRepo with a given path to a local repository.
+        Clone from the given source if a git repository does not exist
+        at the given path or it is broken (i.e., something exists at the
+        path but is not a git repository).
+        Note that the files at the path are cleaned up before cloning in
+        second case.
+
+        Parameters:
+            path (str): Source path of the local git repository.
+            clone_from (str): Url/Path to remote repository to clone from.
+                              (defaults to None)
+            recurse_submodules (bool): Recurse into submodules (default is
+                                       False)
+            shallow_submodules (bool): any cloned submodules will be shallow.
+                                       (default is False)
+            clone_if_broken (bool): Clone from source if local repository looks
+                                    broken. (default is False)
+            clone_if_not_found (bool): Clone from source if local repository is
+                                       not found. (default is False)
+
+        Returns:
+            GitRepo: the GitRepo object for the local git repository.
+
+        Raises:
+            ValueError: If clone_from is not provided and a clone attempt is
+                        made.
+            FileNotFoundError: If local git repository is not found and
+                               clone_if_not_found is disabled.
+            NonGitDirectoryError: If the directory at the given path is not
+                                  a git repository and clone_if_broken is
+                                  disabled.
+        """
+        repo: GitRepo
+        try:
+            repo = GitRepo(
+                local_path=path
+            )
```

Comment:
Should there be a sanity check in here that if the thing exists and a clone-from has been suppliedthat the repo is actually a clone of clone_from?

Reply:
I don't think there is an easy way to tell where the repo was cloned from. We can check the URL of the origin remote, but that can be spoofed as well. I can clone from your remote and then set my remote as the origin. Any such check can easily be bypassed, so I don't think there's any need for it. We can leave it to the user to ensure they've cloned from the right remote.